src/sdg/core/models/managers.py

```diff
@@ -0,0 +1,59 @@
+from copy import deepcopy
+
+from django.db import models
+from django.db.models import BooleanField, Case, Q, Value, When
+from django.utils.timezone import now
+
+
+class ProductenCatalogusQuerySet(models.QuerySet):
+    def specific_catalogs(self, municipality):
+        """
+        Get specific catalogs for the given municipality.
+        Select the reference catalog for extra performance.
+        Annotate whether the reference catalog is owned by the municipality.
+        """
+        return (
+            self.filter(
+                lokale_overheid=municipality,
+                is_referentie_catalogus=False,
+            )
+            .select_related(
+                "referentie_catalogus",
+            )
+            .annotate(
+                municipality_owns_reference=Case(
+                    When(
+                        referentie_catalogus__lokale_overheid=municipality,
+                        then=Value(True),
+                    ),
+                    default=Value(False),
+                    output_field=BooleanField(),
+                )
+            )
+        )
+
+    def annotate_area_and_products(self):
```

Comment:
I think I mentioned I don't want this in a manager due to its cross-link behaviour with the view. It also doesn't annotate any products, despite the function name ;-)

I'll hint in the view what to do.